src/openrct2-ui/interface/Graph.cpp

```diff
@@ -182,10 +183,129 @@ static void graph_draw_line_b_money32(
     }
 }
 
+struct financial_tooltip_info
+{
+    const CoordsXY coords;
+    const money32 money{};
+};
+
+struct Rect
+{
+    const CoordsXY LeftTop;
+    const CoordsXY RightBottom;
+
+    int32_t GetLeft() const
+    {
+        return LeftTop.x;
+    }
+    int32_t GetTop() const
+    {
+        return LeftTop.y;
+    }
+    int32_t GetRight() const
+    {
+        return RightBottom.x;
+    }
+    int32_t GetBottom() const
+    {
+        return RightBottom.y;
+    }
+    int32_t GetWidth() const
+    {
+        return RightBottom.x - LeftTop.x;
+    }
+    int32_t GetHeight() const
+    {
+        return RightBottom.y - LeftTop.y;
+    }
+    bool Contains(const CoordsXY& coords) const
+    {
+        return coords.x >= LeftTop.x && coords.x <= RightBottom.x && coords.y >= LeftTop.y && coords.y <= RightBottom.y;
+    }
+    bool Contains(const ScreenCoordsXY& coords) const
+    {
+        return Contains(CoordsXY(coords.x, coords.y));
+    }
+};
+
+static constexpr auto chartMaxWidth = 63 * 6;
+static constexpr auto chartMaxHeight = 164;
+
+static int32_t indexForCursorAndHistory(const int32_t& historyCount, const int32_t& cursorX, const int32_t& chartX)
+{
+    const auto cursorOffset = 3;
+    const auto offsettedCursorX = cursorX + cursorOffset;
+    return (historyCount - 1) - (offsettedCursorX - chartX) / 6;
+}
+
+static const CoordsXY coordsForHistoryIndex(
+    const int32_t& index, const money32* history, const int32_t& chartX, const int32_t& chartY, const int32_t& modifier,
+    const int32_t& offset)
+{
+    CoordsXY coords;
+    coords.x = chartX + 6 * (63 - index);
```

Comment:
`63` and `6` should be extracted to constants, they're used all over, including on other constants, though I didn't check if it's just a coincidence